doc/src/tutorial/004_full_expressions.md

```diff
@@ -70,6 +68,34 @@ impossible: a `Factor` cannot have an `Expr` as its left-hand side.
 This is the purpose of the tiers: to force the parser into the
 precedence you want.
 
+This can also be accomplished using the `precedence` and `assoc` attribute macros.
+They remove the need to make tiered expressions which 
+reduces code complexity when working with many levels of precedence.
+The above `Expr` grammar can be rewritten to the following using them:
+
+```
+pub Expr: i32 = {
+    #[precedence(level="0")] // Highest precedence
+    Term,
+    #[precedence(level="1")] #[assoc(side="left")]
+    <l:Expr> "*" <r:Expr> => l * r,
+    #[precedence(level="2")] #[assoc(side="left")]
+    <l:Expr> "/" <r:Expr> => l / r,
+    #[precedence(level="3")] #[assoc(side="left")]
+    <l:Expr> "+" <r:Expr> => l + r,
+    #[precedence(level="4")] #[assoc(side="left")]
+    <l:Expr> "-" <r:Expr> => l - r,
+};
+```
+
+The `precedence` level specifies the order of operations starting from zero.
+In this example it means that `13 + 7 * 3` would be evaluated as `(13 + (7 * 3))` 
+because multiplication has a lower `precedence` level than addition.
+
+By using `assoc` you can specify if the expression is left-associative or right-associative.
+This is required to make the grammar unambiguous, otherwise `1 + 2 + 3` could 
+both be interpreted as `(1 + (2 + 3))` and `((1 + 2) + 3)`.
+
```

Comment:
I rewrote the first part to include this since I could not nicely fit in a lone sentence or two at the end about this. If I am missing stuff please suggest changes as I am not knowledgeable enough about the inner workings of parsers to include more detail.